tests/unit/lib/test_file.py

```diff
@@ -379,3 +380,18 @@ def test_get_local_path(tmp_path, catalog):
     assert file.get_local_path() is None
     file.ensure_cached()
     assert file.get_local_path() is not None
+
+
+@pytest.mark.parametrize("use_cache", (True, False))
+def test_export_with_symlink(tmp_path, catalog, use_cache):
+    path = tmp_path / "myfile.txt"
+    path.write_text("some text")
+
+    file = File(path=path.name, source=f"file://{tmp_path}")
+    file._set_stream(catalog, use_cache)
+
+    file.export(tmp_path / "dir", link_type="symlink", use_cache=use_cache)
+    assert (tmp_path / "dir" / "myfile.txt").is_symlink()
+
+    dst = Path(file.get_local_path()) if use_cache else path
+    assert (tmp_path / "dir" / "myfile.txt").resolve() == dst
```

Comment:
Using `resolve()` instead of `readlink()` because `WindowsPath.readlink()` returns an extended path that fails to compare.

```diff
-- WindowsPath('C:/Users/runneradmin/AppData/Local/Temp/pytest-of-runneradmin/pytest-0/popen-gw2/default_working_dir0/.datachain/cache/29/89fe8cd7eda0a1e2e28bf837e137f5bdd8d20c506348f67f5671f5a036d529')
++ WindowsPath('//?/C:/Users/runneradmin/AppData/Local/Temp/pytest-of-runneradmin/pytest-0/popen-gw2/default_working_dir0/.datachain/cache/29/89fe8cd7eda0a1e2e28bf837e137f5bdd8d20c506348f67f5671f5a036d529')
```